upload/upload.go

```diff
@@ -85,6 +86,20 @@ func isLegacyTestRequest(r *http.Request) bool {
 	return false
 }
 
+func isSatelliteTestRequest(r *http.Request) bool {
+
+	if r.Header.Get("Content-Type") == "application/json" {
+		buf := new(bytes.Buffer)
+		buf.ReadFrom(r.Body)
+		body := buf.String()
+		if body == `{"test": "test"}` {
```

Comment:
Performing a check like this could be brittle.  A client might pass in a piece of json that looks like this:
```
{
  "test": "test"
}
```
which would cause the test to return false.  

Maybe checking the string is exactly as expected is ok in this case...as long as Satellite passes the string exactly as is and does not change it?  Just something to watch out for...

Is this the only case where the content-type is set to "application/json"?  I just want to make sure this processing logic doesn't happen when 100+ mb files are uploaded.  :)

Reply:
@dehort I definitely considered that. I didn't want to shoot myself in the foot with a test like this, so I also needed to keep it as close to what we see from a satellite test as possible. it's a narrow scope, and nothing exactly blows up if we say it's "false". It will just get an error that it's missing the "file/upload" field.
I expect we'll accept plain ol' JSON at some point, but we'll need to hack at all this stuff again anyway in that case.